localstack/services/kinesis/kinesis_listener.py

```diff
@@ -319,8 +321,9 @@ def send_events():
                 record["Data"] = to_str(base64.b64encode(record["Data"]))
                 last_sequence_number = record["SequenceNumber"]
             if not records:
-                time.sleep(1)
-                continue
+                # on AWS this is approximately 5 sec, however since this is not async, it's a blocking call
+                # putting temporarily to 3 seconds until ASF migration or an async call
```

Comment:
Could you elaborate on that comment? The `send_events` generator is executed async by the HTTP framework already, isn't it?

Reply:
as discussed